frontend/js/annotation-tool.js

```diff
@@ -685,9 +689,212 @@ define(["jquery",
+                    var view = new Uint8Array(buf);  //create uint8array as viewer
+                    for (var i=0; i<s.length; i++) view[i] = s.charCodeAt(i) & 0xFF; //convert to octet
+                    return buf;    
+                }
+
+                saveAs(new Blob([s2ab(wbout)],{type:"application/octet-stream"}), 'export.xlsx');
+            },
+
+            gatherExportData: function(tracks, categories, freeText) {
+                var bookData = [];
+                var header = [];
+                addResourceHeaders(header);
+                header.push("Track name");
+                header.push("Leadin");
+                header.push("Leadout");
+                header.push("Duration");
+                header.push("Text");
+                header.push("Category name");
+                header.push("Label name");
+                header.push("Label abbreviation");
+                header.push("Scale name");
+                header.push("Scale value name");
+                header.push("Scale value value");
+                addResourceHeaders(header, "comment");
+                header.push("Comment text");
+                header.push("Comment replies to");
+                bookData.push(header);
+
+                _.each(tracks, function (track) {
+                    _.each(annotationTool.getAnnotations(track.id), function (annotation) {
+                        var line = [];
+
+                        let label = annotation.attributes.label;
+                        // No idea what this check is good for tbh
+                        if (label) {
+                            if (categories && !categories.map(category => category.id).includes(label.category.id)) return;
+                        } else {
+                            if (!freeText) return;
+                        }
+                        
+                        addResource(line, annotation)
+                        line.push(track.attributes.name);
+
+                        line.push(util.formatTime(annotation.attributes.start));
+                        line.push(util.formatTime(annotation.attributes.start + annotation.attributes.duration));
+                        line.push(util.formatTime(annotation.attributes.duration));
+                        line.push(annotation.attributes.text);
+
+                        if (label) {
+                            line.push(label.category.name)
+                            line.push(label.value)
+                            line.push(label.abbreviation)
+                        } else {
+                            line.push("")
+                            line.push("")
+                            line.push("")
+                        }
+                        
+                        // What the heck is the difference between 'scalevalue' and 'scaleValue'?
+                        // They seem to contain the exact same data
```

Comment:
Let me introduce you to one of the biggest monstrosities of this codebase: https://github.com/opencast/annotation-tool/blob/master/frontend/js/models/annotation.js#L89-L92

And then later on in the same function: https://github.com/opencast/annotation-tool/blob/master/frontend/js/models/annotation.js#L129-L131

I have no idea why. :sob: 

```suggestion
```

Reply:
...Fascinating, thank you.